packages/components/src/components/stepper-horizontal/stepper-horizontal-item/stepper-horizontal-item-styles.ts

```diff
@@ -74,96 +71,99 @@ export const getComponentCss = (state: StepperState, disabled: boolean, theme: T
             (result, _, i) => ({
               ...result,
               [`&(:nth-of-type(${i + 1})) $button::after`]: {
-                backgroundPositionX: `${-i * spriteStepSize}em`,
+                backgroundImage: getInlineSVGBackgroundImage(
+                  getSVGPath(
+                    i + 1,
+                    {
+                      primaryColor,
+                      invertedBaseColor: getInvertedThemedColors(theme).primaryColor,
+                      disabledColor,
+                    },
+                    isStateCurrent
+                  )
+                ),
               },
             }),
             {} as JssStyle
           )),
         ...addImportantToEachRule({
           fontSize: 'inherit',
+          transform: 'translate3d(0,0,0)', // creates new stacking context
           ...hostHiddenStyles,
           '&(:not(:last-of-type))': {
-            marginRight: '1em',
+            margin: '0 16px 0 0',
           },
         }),
       },
       button: {
+        display: 'grid',
```

Comment:
I think `grid` is the wrong choice here and adds additional overhead with the columns and areas.
Probably the best way would be to apply the background svg on this element, then there wouldn't be any need for the pseudo. The space for the svg can then be reserved via padding.

Reply:
Using the before for the background image so we got natural flex flow with gap